testdata/e2e/e2e_test.go

```diff
@@ -232,6 +279,279 @@ func parseHostFromFileName(fileName string) string {
 	return strings.TrimSuffix(s[2], filepath.Ext(s[2]))
 }
 
+func AsCadvisorMetrics(baseline bool) UnmarshalForK8sListFn {
+	return func(fname string, fdata []byte, parsedK8sList *ParsedK8sLists) error {
+		cPromMetrics, err := parseRawCadvisorPrometheusFile(fdata)
+		if err != nil {
+			return err
+		}
+		cPromMetricsCI := parsedK8sList.CadvisorPrometheus
+		if baseline {
+			cPromMetricsCI = parsedK8sList.BaselineCadvisorPrometheus
+		}
+		cPromMetricsCI[parseHostFromFileName(fname)] = cPromToContainerInfo(cPromMetrics)
+		return nil
+	}
+}
+
+func parseRawCadvisorPrometheusFile(fdata []byte) (map[string]*prom2json.Family, error) {
+	cPromMetrics := make(map[string]*prom2json.Family)
+	in := bytes.NewReader(fdata)
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(in)
+	if err != nil {
+		return nil, fmt.Errorf("reading cadvisor prometheus text format failed: %v", err)
+	}
+	for name, mf := range metricFamilies {
+		cPromMetrics[name] = prom2json.NewFamily(mf)
+	}
+	return cPromMetrics, nil
+}
+
+func cPromToContainerInfo(metricFamilies map[string]*prom2json.Family) map[string]cadvisor.ContainerInfo {
+	containerInfos := make(map[string]cadvisor.ContainerInfo)
+	missingMetrics := make(map[string]bool)
+	for metricName, metricsFamily := range metricFamilies {
+		for _, containerMetric := range metricsFamily.Metrics {
+			metric, ok := containerMetric.(prom2json.Metric)
+			if !ok {
+				continue
+			}
+			container := metric.Labels["id"]
+			if container == "" {
+				container = "node-level-metric"
+			}
+			var ci cadvisor.ContainerInfo
+			var err error
+			if ci, ok = containerInfos[container]; !ok {
+				ci, err = newContainerInfoFromCProm(metric, container)
+				if err != nil {
+					continue
+				}
+			}
+			ci = containerMetricsFromCProm(ci, metricName, metric, missingMetrics)
+			containerInfos[container] = ci
+		}
+	}
+	return containerInfos
+}
+
+func containerMetricsFromCProm(
```

Comment:
Given the hefty amount of code here for converting Cadvisor Prometheus to the ContainerInfo format in addition to the existing utility code, it might be cleaner to split into a new file at this point

Reply:
Good call 👍